src/main/scala/io/circe/yaml/parser/package.scala

```diff
@@ -38,6 +39,15 @@ package object parser {
       None
   }
 
+  private[this] class FlatteningConstructor extends SafeConstructor {
+    def flatten(node: MappingNode): MappingNode = {
+      flattenMapping(node)
+      node
+    }
+  }
+
+  private[this] val flattener: FlatteningConstructor = new FlatteningConstructor
```

Comment:
Are you sure this is threadsafe? SnakeYAML's docs explicitly absolve them of thread safety. It could be that this particular bit of SnakeYAML *is* threadsafe, but maybe it's best to create a new `FlatteningConstructor` in the call to `parse` and pass it through all the `yamlToJson`s just to be sure? It sure seems like it ought to be safe since you're only using it for `flatten`, but `Constructor` does seem to have all kinds of non-threadsafe mutable state, and it wouldn't be trivial to ensure it's impossible for it to get touched by this.